python/lsst/ctrl/bps/htcondor/dagman_configurator.py

```diff
@@ -0,0 +1,219 @@
+# are missing from ``htcondor.param``.
+#
+# Notes
+# -----
+# A complete list of configuration options HTCondor supports can be found in
+# ``src/condor_utils/param_info.in`` in
+# `HTCondor GitHub repository <https://github.com/htcondor/htcondor>`_.
+#
+# It is assumed that these options only change between major LTS releases.
+_ver = version.parse(htcondor.__version__)
+match (_ver.major, _ver.minor):
+    case (24, 0):
+        _fields.update(
+            {
+                "dagman_debug": (str, ""),
+                "dagman_node_record_info": (str, ""),
+                "dagman_record_machine_attrs": (str, ""),
+            }
+        )
+    case (25, 0):
+        _fields.update(
+            {
+                "dagman_config_file": (str, ""),
+                "dagman_debug": (str, ""),
+                "dagman_insert_sub_file": (str, ""),
+                "dagman_node_record_info": (str, ""),
+                "dagman_record_machine_attrs": (str, ""),
+            }
+        )
+
+# Dynamically create a Pydantic model encapsulating the DAGMan configuration
+# options gathered above.
+_DagmanOptions = create_model(
+    "DagmanOptions",
+    __config__=ConfigDict(
+        alias_generator=AliasGenerator(
+            serialization_alias=lambda name: name.upper(),
+        ),
+        extra="allow",
+        serialize_by_alias=True,
+    ),
+    **_fields,
+)
+
+
+class DagmanConfigurator:
+    """Class responsible for setting WMS-specific configuration options.
+
+    Parameters
+    ----------
+    config : `lsst.ctrl.bps.BpsConfig`
+        BPS configuration.
+    search_opts : `dict` [`str`, `Any`], optional
+        Options to use while searching the BPS configuration for values.
+
+    Raises
+    ------
+    KeyError
+        Raised if DAGMan configuration is missing from the BPS configuration.
+    """
+
+    def __init__(self, config: BpsConfig, search_opts: dict[str, Any] | None = None) -> None:
+        if search_opts is None:
+            search_opts = {}
+        _, site = config.search("computeSite", search_opts)
+        if site:
+            search_opts["curvals"] = {"curr_site": site}
+        _, wms_config = config.search("wmsConfig", search_opts)
+        if not wms_config:
+            raise KeyError("WMS-specific configuration not found")
+        self._options = _DagmanOptions.model_validate({key.lower(): val for key, val in wms_config.items()})
+        if self._options.model_extra:
+            unknown_opts = [key.upper() for key in self._options.model_extra]
+            _LOG.warning(
+                "Following WMS-specific configuration options were not recognized and will be ignored: %s.",
+                ", ".join(unknown_opts),
+            )
+        self.config_path: Path | None = None
+        self.prefix: Path | None = None
+
+    @property
+    def options(self) -> dict[str, Any]:
+        """Valid DAGMan configuration options (`dict` [`str`, `Any`])."""
```

Comment:
Do we want this to include unset options?  If yes, please include that fact in the description (or better describe what valid means).  If no, then can add the `exclude_unset=True` and use this property when writing out the config.